src/OpenApi/JSON.hs

```diff
@@ -0,0 +1,89 @@
+module OpenApi.JSON
+  ( generateRenamed
+  , parseJSONFixed
+  , parseRefSum
+  , parseRenamed
+  )
+where
+
+import Data.Map.Strict (Map)
+import Data.String (String)
+import GHC.Enum (Bounded, Enum, minBound)
+import GHC.Generics (Generic, Rep)
+import OpenApi.Prelude
+
+import qualified Data.Aeson.Types    as JSON
+import qualified Data.HashMap.Strict as HashMap
+import qualified Data.List           as List
+import qualified Data.Map.Strict     as Map
+import qualified Data.Text           as Text
+
+parseRenamed
+  :: (Generic a, JSON.GFromJSON JSON.Zero (Rep a))
+  => Map String String
+  -> JSON.Value
+  -> JSON.Parser a
+parseRenamed = JSON.genericParseJSON . renameOptions
+
+generateRenamed
+  :: (Generic a, JSON.GToJSON JSON.Zero (Rep a))
+  => Map String String
+  -> a
+  -> JSON.Value
+generateRenamed = JSON.genericToJSON . renameOptions
+
+parseJSONFixed
+  :: forall a b c . (Bounded b, Enum b, Eq a, Show a)
+  => Text
+  -> (String -> (a -> JSON.Parser b) -> c -> JSON.Parser b)
+  -> (b -> a)
+  -> c
+  -> JSON.Parser b
+parseJSONFixed name withValue map =
+  withValue (convertText name) $ \value ->
+    maybe (unexpected value) pure $ List.lookup value options
+  where
+    options :: [(a, b)]
+    options = (\item -> (map item, item)) <$> [minBound..]
```

Comment:
You could write this as:

```haskell
options = ((,) =<< map) <$> [minBound..]
```

Reply:
I see that this is a suggestion from a time before you replaced `(String, String)` with `Map String String`. Marking this as resolved.

Reply:
Yes, this was a suggestion that applied to the code as it was at the time of my review. I do not think it applies to the "new" Map version.